cpp/oneapi/dal/table/backend/common_kernels.hpp

```diff
@@ -70,8 +70,8 @@ inline bool alloc_kind_requires_copy(alloc_kind src_alloc_kind, alloc_kind dst_a
 #else
     ONEDAL_ASSERT(src_alloc_kind == alloc_kind::host);
     ONEDAL_ASSERT(dst_alloc_kind == alloc_kind::host);
-    return false;
 #endif
+    return false;
```

Comment:
Does it better to throw exaptation with error like "error_msg::unsupported_alloc_kind()" or so (in line 68, and revert return in previous place)? We have similar situation with alloc_kind_from_sycl(...) or alloc_kind_to_sycl(...) and throw an exaptation.